lib/elixir/lib/registry.ex

```diff
@@ -640,7 +668,7 @@ defmodule Registry do
     # Remove first from the key_ets because in case of crashes
     # the pid_ets will still be able to clean up. The last step is
     # to clean if we have no more entries.
-    true = :ets.match_delete(key_ets, {key, {self, :_}})
+    true = :ets.match_delete(key_ets, {key, {self, pattern}})
     true = :ets.delete_object(pid_ets, {self, key, key_ets})
```

Comment:
I'm not sure how we get around this issue without modifying the shape of the `pid_ets` table entries.

If the key_ets rows had a unique integer in it or something that could also be in the pid_ets entry you coudl do a match first on the key ets, get the unique integers you were going to delete, then do a delete_match on the key ets, and a delete_match on the pid ets with the unique inters.